platform/api/azure/api.go

```diff
@@ -55,5 +90,42 @@ func New(opts *Options) (*API, error) {
 		opts:   opts,
 	}
 
+	err = api.resolveImage()
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve image: %v", err)
+	}
+
 	return api, nil
 }
+
+func randomName(prefix string) string {
+	b := make([]byte, 5)
+	rand.Read(b)
```

Comment:
Should be `ReadFull()` (see https://github.com/satori/go.uuid/issues/73 as an example why)